test/csuite/wt8659_reconstruct_database_from_logs/main.c

```diff
@@ -0,0 +1,330 @@
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s && mkdir %s", dir, dir));
+    testutil_check(system(buf));
+}
+
+/*
+ * remove_dir --
+ *     Removes the directory.
+ */
+static void
+remove_dir(const char *dir)
+{
+    char buf[1024];
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s", dir));
+    testutil_check(system(buf));
+}
+
+/*
+ * compare_backups --
+ *     Compares the full and the incremental backups.
+ */
+static int
+compare_backups(void)
+{
+    int ret;
+    char buf[1024];
+
+    /*
+     * We have to copy incremental backup to keep the original database intact. Otherwise we'll get
+     * "Incremental backup after running recovery is not allowed".
+     */
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "cp %s/* %s", home_incr, home_incr_copy));
+    testutil_check(system(buf));
+
+    /* Dump both backups. */
+    dump_table(home_full, uri, full_out);
+    dump_table(home_incr_copy, uri, incr_out);
+
+    reset_dir(home_incr_copy);
+
+    /* Compare the files. */
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "cmp %s %s", full_out, incr_out));
+    if ((ret = system(buf)) != 0) {
+        printf(
+          "Tables \"%s\" don't match in \"%s\" and \"%s\"!\n See \"%s\" and \"%s\" for details.\n",
+          uri, home_full, home_incr_copy, full_out, incr_out);
+        exit(1);
+    } else {
+        /* If they compare successfully, clean up. */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm %s %s", full_out, incr_out));
+        testutil_check(system(buf));
+        printf("\t Table \"%s\": OK\n", uri);
+    }
+
+    return (ret);
+}
+
+/*
+ * add_work --
+ *     Inserts some data into the database.
+ */
+static void
+add_work(int iter)
+{
+    WT_CURSOR *cursor;
+    int i;
+    char k[32], v[128];
+
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
+    /* Perform some operations with individual auto-commit transactions. */
+    for (i = 0; i < MAX_KEYS; i++) {
+        testutil_check(__wt_snprintf(k, sizeof(k), "key.%d.%d", iter, i));
+        testutil_check(__wt_snprintf(v, sizeof(v), "value.%d.%d", iter, i));
+        cursor->set_key(cursor, k);
+        cursor->set_value(cursor, v);
+        testutil_check(cursor->insert(cursor));
+    }
+    testutil_check(cursor->close(cursor));
+}
+
+/*
+ * take_full_backup --
+ *     Takes full backup of the database.
+ */
+static void
+take_full_backup(const char *home, const char *backup_home)
+{
+    WT_CURSOR *cursor;
+    int ret;
+    char buf[1024];
+    const char *filename;
+
+    testutil_check(session->open_cursor(session, "backup:", NULL, NULL, &cursor));
+
+    while ((ret = cursor->next(cursor)) == 0) {
+        testutil_check(cursor->get_key(cursor, &filename));
+        testutil_check(
+          __wt_snprintf(buf, sizeof(buf), "cp %s/%s %s/%s", home, filename, backup_home, filename));
+        testutil_check(system(buf));
+    }
+
+    scan_end_check(ret == WT_NOTFOUND);
```

Comment:
Can we use `testutil_assert` isntead?

Reply:
It expands into `testutil_assert`. `scan_end_check` name is more clear in this instance.

Reply:
It seems that it is only used in the `examples/c` files and `import.c` though. @sueloverso , what is your opinion?

Reply:
If you look at the comment for `scan_end_check` in `test_util.h` it says `The same as testutil_assert, with a different name because it appears in the documentation.` The original code is in the examples and therefore appears in the documentation. Since this is now a test, I think it should use `testutil_assert`.

Reply:
OK, done.